flowredux/src/commonMain/kotlin/com/freeletics/flowredux/dsl/ConditionBuilderBlock.kt

```diff
@@ -7,4 +7,14 @@ import kotlinx.coroutines.ExperimentalCoroutinesApi
 @FlowReduxDsl
 public class ConditionBuilderBlock<InputState : S, S : Any, A : Any> internal constructor(
     override val isInState: SideEffectBuilder.IsInState<S>,
-) : BaseBuilderBlock<InputState, S, A>()
+) : BaseBuilderBlock<InputState, S, A>() {
+
+    public fun untilIdentityChanges(
+        identity: (InputState) -> Any,
```

Comment:
Should we add some type alias (IndentityCheck) with some proper java doc that identity is checked with `==` operator? Typealias just to not have to repeat the javadoc in `InStateBuilderBlock` and in `ConditionBuilderBlock` (but maybe without typealias and just copy & pase javadoc is doing the job too since it is only 2 places). 



I am also wondering if we should define a lambda instead where one could specify how identity works? What I mean by that is a way to compare to states (previous and current state) if they have the same identity: `checkSameIdentity: (InputState, InputState) -> Boolean` With that way one could also use other operators to compare identity, i.e. `===` like `checkSameIdentity = { previous : Inputstate , current : IntputState -> previous.foo === current.foo }`. Mabye it is an overkill. Just thinking out loud ...

Reply:
The more I think about it, the more I think it is overkill to have `checkSameIdentity: (InputState, InputState) -> Boolean` 😄 .


Probably  this would be another operator something like `untilChanged` or whatever. But I think, until there is no usecase, we can keep your proposal